test/integration/buffer_accounting_integration_test.cc

```diff
@@ -396,4 +405,105 @@ TEST_P(ProtocolsBufferWatermarksTest, ResettingStreamUnregistersAccount) {
   }
 }
 
+class Http2OverloadManagerIntegrationTest : public Http2BufferWatermarksTest,
+                                            public Envoy::BaseOverloadIntegrationTest {
+protected:
+  void initializeOverloadManagerInBootstrap(
+      const envoy::config::overload::v3::OverloadAction& overload_action) {
+    setupOverloadManagerConfig(overload_action);
+    overload_manager_config_.mutable_buffer_factory_config()
+        ->set_minimum_account_to_track_power_of_two(absl::bit_width(4096u));
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      *bootstrap.mutable_overload_manager() = this->overload_manager_config_;
+    });
+  }
+};
+
+// Run the tests using HTTP2 only since its the only protocol that's fully
+// supported.
+// TODO(kbaichoo): Instantiate with H3 and H1 as well when their buffers are
+// bounded to accounts.
+INSTANTIATE_TEST_SUITE_P(
+    IpVersions, Http2OverloadManagerIntegrationTest,
+    testing::Combine(testing::ValuesIn(HttpProtocolIntegrationTest::getProtocolTestParams(
+                         {Http::CodecType::HTTP2}, {FakeHttpConnection::Type::HTTP2})),
+                     testing::Bool()),
+    protocolTestParamsAndBoolToString);
+
+TEST_P(Http2OverloadManagerIntegrationTest, ResetsExpensiveStreamsWhenOverloaded) {
```

Comment:
can we make sure to have an integration test where upstream buffers take too much space and another where downstream buffers do?